src/os.cc

```diff
@@ -266,6 +260,13 @@ long long file::size() const {
 #  endif
 }
 
+// Return type of read and write functions.
+#  ifdef _WIN32
+#    define rwresult int
+#  elif FMT_USE_FCNTL
+#    define rwresult ssize_t
+#  endif
```

Comment:
What is this for?

Reply:
It's a way to fix this error:

```
In file included from /mnt/d/dev/libs/tools/fmt/src/fmt.cc:148:
/mnt/d/dev/libs/tools/fmt/src/os.cc:74:7: error: ‘using {anonymous}::rwresult = ssize_t’ exposes TU-local entity ‘{anonymous}’
   74 | using rwresult = ssize_t;
      |       ^~~~~~~~
/mnt/d/dev/libs/tools/fmt/src/os.cc:62:1: note: ‘{anonymous}’ declared with internal linkage
   62 | namespace {
      | ^~~~~~~~~
```

When using `using`,  `rwresult` becomes a TU-local type and has internal linkage because it's in anonymous namespace.

Reply:
Typedefs are just aliases and shouldn't have linkage of their own. And even if they had we wouldn't want to replace them with macros. Please revert.

Reply:
My understanding is that typedefs inherit the linkage of the type they refer to. This means that the name `rwresult` has linkage because - in that case - it refers to `ssize_t`, which is a fundamental type and therefore has linkage itself (see [basic.link](https://timsong-cpp.github.io/cppwp/n4659/basic.link#8)).

I initially thought of 2 ways to work around this (maybe there are other ways):

- take the `using rwresult = <type>` out of the anonymous namespace, but then this exposes `rwresult`, which was hidden previously so I elected not to do that
- use a macro which can be `#undef`ed hence not exposing the `rwresult` name

I'm happy to take `rwresult` out of the anonymous namespace if you don't mind exposing `rwresult`, unless there is a more elegant solution.

Reply:
I'm not sure gcc is correct here - it even seems to be inconsistent not just with other compilers but with itself since there are other TU-local symbols here like `convert_rwcount`. I suggest reverting this change and possibly opening an issue in gcc to clarify why it complains for some symbols and not the other.

Reply:
Yes, indeed, it's weird that `convert_rwcount` doesn't raise anything. Enquiry sent on gcc-help, I'll keep you posted.